packages/react-components/src/components/Datepicker/helpers.ts

```diff
@@ -0,0 +1,144 @@
+import {
+  isAfter,
+  isSameDay,
+  isSameMonth,
+  subMonths,
+  differenceInCalendarDays,
+} from 'date-fns';
+import { Modifiers } from 'react-day-picker';
+import {
+  IRangeDatePickerProps,
+  IRangeDatePickerState,
+  IRangeDatePickerOption,
+} from './types';
+
+const baseClass = 'date-picker';
+
+import styles from './DatePicker.module.scss';
+
+export const isDateWithinRange = (
+  date: Date,
+  range: { from?: Date; to?: Date }
+): boolean => {
+  const { from, to } = range;
+  if (to && !isSameDay(date, to) && isAfter(date, to)) {
+    return false;
+  }
+  if (from && !isSameDay(date, from) && !isAfter(date, from)) {
+    return false;
+  }
+  return true;
+};
+
+export const calculateDatePickerMonth = (
+  initialFromDate?: Date,
+  initialToDate?: Date,
+  toMonth?: Date
+): Date => {
+  if (initialToDate) {
+    const forcePreviousMonth =
+      initialFromDate && !isSameMonth(initialFromDate, initialToDate);
+
+    if (
+      forcePreviousMonth ||
+      (toMonth && isSameMonth(initialToDate, toMonth))
+    ) {
+      return subMonths(initialToDate, 1);
+    }
+
+    return initialToDate;
+  }
+
+  return subMonths(toMonth || new Date(), 1);
```

Comment:
same as above

Reply:
I don't know how I can simplify this. 
we have to return a month that will be marked as a "current" one. Cases I see here are as follows:
 - if nothing has been provided to the helper, return last month
 - if only toMonth was supplied return it as the current month
 - if initialFromDate and initialToDate are different return last month.
 - if initialFromDate is not available and toMonth is not available return initialToDate
 Can someone help me out with this?

Reply:
```suggestion
  if (!initialToDate) {
     return subMonths(toMonth || new Date(), 1);
  }
  
  const forcePreviousMonth =  initialFromDate && !isSameMonth(initialFromDate, initialToDate);

  if (forcePreviousMonth || (toMonth && isSameMonth(initialToDate, toMonth))) {
    return subMonths(initialToDate, 1);
  }

  return initialToDate;
```

Reply:
👌🏼